src/components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactElement, useCallback, useEffect, useRef, useState } from 'react'
+import classNames from 'classnames'
+
+import { computeItems, renderItem } from './Breadcrumb.utils'
+import { BreadcrumbButton } from './Breadcrumb.types'
+import { BreadcrumbProps } from './Breadcrumb.props'
+import styles from './Breadcrumb.module.css'
+
+/**
+ * UI component for displaying the current location within an hierarchy.
+ *
+ * @returns {Breadcrumb} Breadcrumb component
+ */
+export const Breadcrumb = ({ isLoading, items, getPopupContainer }: BreadcrumbProps): ReactElement => {
+  const [visibleItems, setVisibleItems] = useState<BreadcrumbButton[]>([])
+
+  const breadcrumbRef = useRef<HTMLDivElement>(null)
+  const hiddenContainerRef = useRef<HTMLDivElement>(null)
+
+  const getDropdownContainer = useCallback(() => {
+    /* istanbul ignore next */
```

Comment:
It is an edge case that should never happen and is not reproducible
The container is always mounted, therefore the ref should always be defined at this point. However, for type safety (and to avoid weird underlying behaviour breaking the UI), I prefer to check rather than cast.